src/js/mqtt.js

```diff
@@ -283,12 +288,16 @@ MqttClient.prototype.disconnect = function(err) {
  */
 MqttClient.prototype.publish = function(topic, payload, options, callback) {
   callback = callback || noop;
+  if (typeof options === 'function') {
+    callback = options;
+  }
```

Comment:
This is unrelated, and it changes the API, revert this, please.

Reply:
Okay.